drivers/wifi/esp32/Kconfig.esp32

```diff
@@ -2,15 +2,11 @@
 
 menuconfig WIFI_ESP32
 	bool "ESP32 SoC WiFi support"
-	select POSIX_API
```

Comment:
Isn't this possible only after updating `hal_espressif`? If yes, then commits are in wrong order and break bisectability.

Also "HAL library does no need posix calls by default" could be elaborated. Posix were never needed? Or the recent HAL update allows to drop posix calls?

Reply:
@mniestroj change the commit order. Thanks.
Yes, HAL was update so that POSIX API is not a dependency anymore.

Reply:
Would be good to have "HAL was updated so that POSIX API is not a dependency anymore." in a commit message :)

Reply:
sure, no problem.